### DISCUSSION TITLE
macOS: Paste the current selection with the right mouse button?

### DISCUSSION THREAD

**Discussion by tjatjagithub** [raphamorim/rio#749] [Q&A]
I am coming from Alacritty and am tempted to replace it with Rio on my Mac.

In Alacritty, I have a configuration that allows to **select text with the left mouse button** and **paste it with the right mouse button** - super standard and fast to use.

As Rio is based on Alacritty, I created those lines for ~/.config/rio/config.toml:

```
[selection]
semantic_escape_chars = ",│`|:\"' ()[]{}<>\t$/="
save_to_clipboard = true

[shell]
program = "/opt/local/bin/bash"
args = ["--login"]

[mouse_bindings]
buttons = [ { mouse = "Right",                 action = "PasteSelection" } ]
```

This does not give a syntax error, but it does not work :-(

The selection cannot be pasted with the right mouse button.

Did I do this wrong, or can Rio not do this?

Any help is appreciated :-)
Many thanks!

**Comment by tjatjagithub** [comment]
Nobody knows, whether it is possible to bind the Right mouse button to Basic Actions like PasteSelection and how to do this?

Please help :-)

**Comment by raphamorim** [comment]
Hey @tjatjagithub this configuration 

```
[selection]
semantic_escape_chars = ",│`|:\"' ()[]{}<>\t$/="
save_to_clipboard = true
```

does not work on rio yet, so you want to select and copy to clipboard right?

**Reply by tjatjagithub** [reply]
Oh, the maintainer himself :-)
Many thanks!

Yes, I just want to copy whatever I selected with the mouse and then paste it with the right mouse button elsewhere.

No chance?

That seems to me such a basic feature - I mean copy-paste is what I do all of the day between windows.
Without this, i simply cannot use a terminal App.
And I seemed to have seen that it is already possible for the middle mouse button! But I cannot use the middle button for this, as it is a wheel and not easy to click with.

Your App can do sooooo many things, and I see that bindings exist and also the Basic Action "PasteSelection", so I thought that I just use the wrong syntax ...
Sad thing :-(

Thanks again!

**Comment by raphamorim** [comment]
Hey there 👋 

Yea that's possible to do, I will convert this discussion into an issue and will take a look on this for the next version.

**Comment by raphamorim** [comment]
https://github.com/raphamorim/rio/discussions/749

**Comment by philippswoboda** [comment]
Hi there! 
Is there any update with the copy/paste functionality with the right mouse button? 🤔 (I'm currently using Windows as my working machine and this a bit of a pain, that it can't be configured yet)
